msprime/intervals.py

```diff
@@ -214,66 +219,24 @@ def uniform_map(cls, length, rate, num_loci=None):
-    # but use the other definition.
     @classmethod
     def read_hapmap(cls, filename):
         """
-        Parses the specified file in HapMap format. These files must be
-        white-space-delimited, and contain a single header line (which is
-        ignored), and then each subsequent line contains the starting position
-        and recombination rate for the segment from that position (inclusive)
-        to the starting position on the next line (exclusive). Starting
-        positions of each segment are given in units of bases, and
-        recombination rates in centimorgans/Megabase. The first column in this
-        file is ignored, as are additional columns after the third (Position is
-        assumed to be the second column, and Rate is assumed to be the third).
-        If the first starting position is not equal to zero, then a
-        zero-recombination region is inserted at the start of the chromosome.
-
-        A sample of this format is as follows::
-
-            Chromosome	Position(bp)	Rate(cM/Mb)	Map(cM)
-            chr1	55550	        2.981822	0.000000
-            chr1	82571	        2.082414	0.080572
-            chr1	88169	        2.081358	0.092229
-            chr1	254996	        3.354927	0.439456
-            chr1	564598	        2.887498	1.478148
-            ...
-            chr1	182973428	2.512769	122.832331
-            chr1	183630013	0.000000	124.482178
+        Parses the specified file in HapMap format.
+
+        .. warning::
+            This method is deprecated, use the module-level
+            :func:`read_hapmap` function instead.
 
         :param str filename: The name of the file to be parsed. This may be
             in plain text or gzipped plain text.
         """
-        positions = []
-        rates = []
-        if filename.endswith(".gz"):
-            f = gzip.open(filename)
-        else:
-            f = open(filename)
-        try:
-            # Skip the header line
-            f.readline()
-            for j, line in enumerate(f):
-                pos, rate, = map(float, line.split()[1:3])
-                if j == 0:
-                    map_start = pos
-                    if pos != 0:
-                        positions.append(0)
-                        rates.append(0)
-                positions.append(pos)
-                # Rate is expressed in centimorgans per megabase, which
-                # we convert to per-base rates
-                rates.append(rate * 1e-8)
-            if rate != 0:
-                raise ValueError(
-                    "The last rate provided in the recombination map must be zero"
-                )
-        finally:
-            f.close()
-        return cls(positions, rates, map_start=map_start)
+        warnings.warn(
+            "RecombinationMap.read_hapmap() is deprecated. "
+            "Use msprime.read_hapmap() instead.",
+            DeprecationWarning,
+        )
+        return read_hapmap(filename)
```

Comment:
So, this would need to ``return RecombinationMap(rate_map.position, list(rate_map.rate) + [0])`` so we've got the right types and don't break old code.